app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -685,9 +687,18 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         return super.onKeyDown(keyCode, event)
     }
 
-    final override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
+    final override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
         if (shouldUseCustomBackLongPress() && keyCode == KeyEvent.KEYCODE_BACK) {
             backLongPressJob?.cancel()
+
+            // check if the key has been pressed for longer than the time needed for a press to turn into a long press
```

Comment:
Could you also please add the context for this speculative fix to this comment.

Reply:
The context is added to the return true value below. We do not want to dismiss an already displayed fragment.

Reply:
Ah ok. Maybe just relocate that comment and combine it with the one above then?

Reply:
Done.